src/mono/netcore/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs

```diff
@@ -425,6 +431,12 @@ public override Assembly GetSatelliteAssembly(CultureInfo culture, Version? vers
 
         public override FileStream[] GetFiles(bool getResourceModules)
         {
+            if (Location == "")
```

Comment:
Nit: Maybe file an issue that this code should move to the shared corelib - for example the GetFiles is basically identical (just different coding style a bit).

Reply:
/cc @CoffeeFlux

Reply:
Some of this should be able to move, just have to keep in mind that changing the layout is hard because we want to match the legacy mono BCL in some cases (or at least, having them differ would result in a bunch of netcore-specific runtime code).